In [1]:
import pandas as pd
import numpy as np
import pandas as pd

file_path = "F23_pre_post_cleaned/F23_pre_post_cleaned.csv"
df = pd.read_csv(file_path, encoding='ISO-8859-1')  # or 'Windows-1252'


In [2]:
df.drop(columns=['ID',\
                 'pre_After reading the participant information on Quercus, please select one of the following:'], inplace=True)


In [3]:
# df.columns.tolist()

In [4]:
df.rename(columns={
    'pre_Are you a first-year student?': 'first_year',
    'pre_Are you an international student?': 'international',
    'pre_Is English your first language?': 'english_first_language',
    'pre_Please tick all that apply. You are:': 'races',
    'pre_What first-year calculus course have you taken or are you currently taking?': 'calculus_course',
    'pre_Have you taken or are you taking a first-year programming course?': 'programming_course',
    'pre_In what statistics program of study have you enrolled or would you like to enrol in (select your first choice)?': 'statistics_program',
    'pre_How would you describe your professional writing ability?': 'pre_professional_writing_ability',
    'pre_What is your experience with programming?': 'pre_programming_experience',
    'pre_What is your experience with the Python programming language?': 'pre_python_experience',
    'pre_How do you evaluate your skill level at Python programming?': 'pre_python_skill_level',
    'pre_How effectively do you feel you can perform statistical data analysis with Python?': 'pre_stats_efficiency_in_python',
    'pre_Which of the following statements best describe your feelings about the use of Python for STA130? Select all that apply.': 'pre_feeling_for_python_sta130',
    'pre_Which of the following statements best describe your feelings about learning to code in Python in STA130? Select all that apply.': 'pre_feeling_learning_python_sta130',
    'pre_Which of the following statements best describes your feelings about learning to code in general?': 'pre_feeling_learning_to_code_general',
    'pre_participation_indicator': 'participation_indicator',
    'post_Approximately how many total hours per week on average did you spend on STA130 this semester?': 'hours_per_week_sta130',
    'post_Approximately how many total hours per week on average did you spend completing STA130 assignments (not counting general studying or attending or reviewing class meetings)?': 'hours_per_week_assignments',
    'post_Which of the following did you routinely utilize and find helpful in supporting your ability to learn Python as part of STA130? (You may select more than one)': 'helpful_tool_python_sta130',
    'post_Which of the following did you find to be the MOST helpful and supportive in terms of performing well in STA130 overall? (You may select more than one)': 'helpful_tool_sta130_general',
    'post_Which aspects of STA130 do you think were the LEAST helpful and supportive for student success in the class? (You may select more than one)': 'least_helpful_tool',
    'post_How do you evaluate your skill level at Python programming?': 'post_python_skill_level',
    'post_How successful were you in learning Python this semester?': 'post_success_learning_python',
    'post_How much has your skill at Python programming increased since the beginning of the semester?': 'post_increase_python_skill',
    'post_Which of the following statements best describes your feelings about learning to code in general?': 'post_feelings_learning_code_general',
    'post_Which of the following statements best describes your assessment of the amount of resources available in STA130 to support your ability to learn to code in Python?': 'assessment_coding_resources_sta130',
    'post_Has STA130 jump-started and fostered your ability to learn python?': 'sta130_jump_started_python',
    'post_Is it helpful to have previous experience with computer programming or simultaneously take a computer programming course to do well in STA130?': 'programming_experience_required',
    'post_Did you enjoy learning statistical and data science concepts based on first introducing them through Python?': 'enjoyed_stats_python',
    'post_Through your experience with the course project and other course (or non-course) activities, how effectively do you feel you can perform statistical data analysis with Python?': 'effective_stats_analysis_python',
    'post_How much do you feel your skill at performing statistical data analysis with Python has improved since the beginning of the semester?': 'improvement_stats_skill_python',
    'post_How strong is your sense of community and belonging here at UofT?': 'community_belonging_uoft',
    'post_How strong is your sense of community and belonging in STA130?': 'community_belonging_sta130',
    'post_How helpful has STA130 been in helping you develop a sense of community and belonging at UofT?': 'sta130_helped_community_belonging',
    'post_What aspects of the STA130 course contributed to and supported developing a sense of community and belonging here at UofT? (You may select more than one)': 'sta130_community_support_aspects',
    'post_How would you describe your professional writing ability?': 'post_professional_writing_ability',
    'post_How helpful were the weekly writing assignments (in the Weekly Homeworks and Tutorial Activities and Assignments) for the development of your professional writing ability?': 'writing_assignments_helped_professional_writing',
    'post_How helpful was the course project and related activities and assignments in giving you an opportunity to apply your communication skills?': 'project_helped_communication_skills',
    'post_What aspects of the course contributed MOST to the development of your professional writing ability? (You may select more than one)': 'course_aspects_helped_professional_writing',
    'post_Describe the purpose of statistical inference (for example, testing and confidence intervals)': 'describe_statistical_inference_purpose',
    'post_Applying statistical methods for description, prediction and inference': 'apply_statistical_methods',
    'post_Visualizing data': 'visualizing_data',
    'post_Use simulation to solve a variety of problems with data': 'use_simulation',
    'post_Manipulating and "wrangling" data': 'manipulating_wrangling_data',
    'post_Using Python for data analysis': 'using_python_data_analysis',
    'post_Communicating in writing to both statisticians and non-statisticians': 'communicating_in_writing',
    'post_Communicating orally to both statisticians and non-statisticians': 'communicating_orally',
    'post_Translating practical problems into statistical terms and vice versa': 'translate_problems_statistical_terms',
    'post_Describing how ethical practice is important in statistical work': 'ethical_practice_importance',
    'post_Using statistical knowledge to solve problems in other disciplines': 'using_stats_other_disciplines',
    'post_Understanding statistical analysis as a problem-solving process (i.e. from study design through data exploration and analysis through communication of results)': 'stats_problem_solving_process',
    'post_Evaluating strengths and limitations of solutions to statistical problems': 'evaluating_stats_solutions',
    'post_What was the most interesting part of this course?': 'most_interesting_part',
    'post_What would you add or what would you change about the course?': 'suggested_course_changes'
}, inplace=True)


In [5]:
df_2023_short_col_name = df

# df_2023_short_col_name.head()
# now 2023 data is ready 

In [6]:
orientation = pd.read_csv('weekly_surveys_raw/STA130_2024_orientation_survey.csv')
wk02 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_02_survey.csv')
# orientation.head()

In [7]:
# orientation.columns


In [8]:
# wk02.columns


In [9]:
orientation.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','0.0','n correct', 'n incorrect','score'], inplace=True)
wk02.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','n correct', 'n incorrect','score'], inplace=True)


In [10]:
orientation.columns

Index(['4389717: Orientation Question 01\r\nWhich AI ChatBots do you use regularly?',
       '4489823: Orientation Question 01 (follow up)\r\nIf you answered "Other" above, which (and perhaps why, if you'd like to share)?',
       '4489822: Orientation Question 02\r\nHow much do you rely on AI ChatBots?',
       '4389718: Orientation Question 03\r\nHave you ever used ChatBot to help with coding?',
       '4421347: Orientation Question 04\r\nDo you think a ChatBot could help support someone doing the kind of statistical and data science analyses taught in STA130?',
       '4389719: Orientation Question 05\r\nDo you think a ChatBot could contribute significantly to your learning process and experience in STA130?',
       '4389725: Orientation Question 06\r\nWhat is your main concern about taking STA130?'],
      dtype='object')

In [11]:
orientation.rename(columns={
    orientation.columns[0]: 'which_chatbots',
    orientation.columns[1]: 'other_chatbots',
    orientation.columns[2]: 'rely_on_chatbot',
    orientation.columns[3]: 'chatbot_for_coding',
    orientation.columns[4]: 'agree_chatbot_help_sta130',
    orientation.columns[5]: 'agree_chatbot_help_you',
    orientation.columns[6]: 'concerns'
}, inplace=True)


# orientation.head()

In [12]:
wk02.columns

Index(['4389967: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '4489824: Question 01 (follow up)\r\nIf you answered "Other" above, which do you use (and perhaps why, if you'd like to share)?',
       '4421351: Question 02 (Week 02)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n"My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '4389974: Question 03 (Accuracy)\r\nDid the ChatBot provide helpful guidance and working code?',
       '4389979: Question 04 (Troubleshooting)\r\nIs the ChatBot able to fix programming problemsin two 

In [13]:
wk02.rename(columns={
    wk02.columns[0]: 'which_chatbots',
    wk02.columns[1]: 'other_chatbots',
    wk02.columns[2]: 'feedback_target',
    wk02.columns[3]: 'accuracy',
    wk02.columns[4]: 'troubleshooting',
    wk02.columns[5]: 'focus',
    wk02.columns[6]: 'additional_feedback'
}, inplace=True)

# wk02.head()

In [14]:
wk03 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_03_survey.csv')
# wk03.columns

In [15]:
wk03.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','n correct', 'n incorrect','score'], inplace=True)
# wk03.columns

In [16]:
wk03.rename(columns={
    wk03.columns[0]: 'which_chatbots',
    wk03.columns[1]: 'other_chatbots',
    wk03.columns[2]: 'feedback_target',
    wk03.columns[3]: 'five_star_scale',
    wk03.columns[4]: 'learning_support',
    wk03.columns[5]: 'focus',
    wk03.columns[6]: 'additional_feedback'
}, inplace=True)

# wk03.head()

In [17]:
wk04 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_04_survey.csv')
wk04.columns

Index(['section', 'section_id', 'section_sis_id', 'submitted', 'attempt',
       '4390022: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '1.0',
       '4489827: Question 01 (follow up)\r\nIf you answered "Other" above, which (and perhaps why, if you'd like to share)?',
       '1.0.1',
       '4390090: Question 02 (Week 04)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n    "My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '1.0.2',
       '4390026: Question 03 (Accuracy)\r\nDid the ChatBot provide helpful guidance and working

In [18]:
wk04.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','n correct', 'n incorrect','score'], inplace=True)

wk04.rename(columns={
    wk04.columns[0]: 'which_chatbots',
    wk04.columns[1]: 'other_chatbots',
    wk04.columns[2]: 'feedback_target',
    wk04.columns[3]: 'accuracy',
    wk04.columns[4]: 'troubleshooting',
    wk04.columns[5]: 'focus',
    wk04.columns[6]: 'additional_feedback'
}, inplace=True)


# wk04.head()



In [19]:
wk05= pd.read_csv('weekly_surveys_raw/STA130_2024_week_05_survey.csv')
# wk05.columns

In [20]:
wk05.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','n correct', 'n incorrect','score'], inplace=True)
# wk05.columns

In [21]:
wk05.rename(columns={
    wk05.columns[0]: 'which_chatbots',
    wk05.columns[1]: 'feedback_target',
    wk05.columns[2]: 'five_star_scale',
    wk05.columns[3]: 'learning_support',
    wk05.columns[4]: 'focus',
    wk05.columns[5]: 'additional_feedback'
}, inplace=True)


# wk05.head()

In [22]:
wk05_after_midterm= pd.read_csv('weekly_surveys_raw/STA130_2024_week_05_after_midterm_survey.csv')
# wk05_after_midterm.columns

In [23]:
wk05_after_midterm.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','n correct', 'n incorrect','score'], inplace=True)
wk05_after_midterm.columns

Index(['4628602: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '4628604: Question 02 (Week 05)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n"My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '4628605: Question 03 (Five-Star Rating)\r\nRate your experience using a ChatBot in STA130 on a Five-Star scale (where 1 is negative and 5 is positive).\r\nDid you have a good experience regarding ChatBot support in writing, using, and troubleshooting Python code; and, learning statistical, data science, and Python programming topics an

In [24]:
wk05_after_midterm.rename(columns={
    wk05_after_midterm.columns[0]: 'which_chatbots',
    wk05_after_midterm.columns[1]: 'feedback_target',
    wk05_after_midterm.columns[2]: 'five_star_scale',
    wk05_after_midterm.columns[3]: 'learning_support',
    wk05_after_midterm.columns[4]: 'focus',
    wk05_after_midterm.columns[5]: 'additional_feedback'
}, inplace=True)

# wk05_after_midterm.head()

In [25]:
wk07= pd.read_csv('weekly_surveys_raw/STA130_2024_week_07_survey.csv')
# wk07.columns

In [26]:
wk07.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','1.0.7','1.0.8','n correct', 'n incorrect','score'], inplace=True)
wk07.columns

Index(['4644490: Question 01 (follow up)\r\nIf you answered "Other" above, which (and perhaps why, if you'd like to share)?',
       '4390110: Question 01\r\nWhich ChatBot(s) are you recently finding helpful or generally useful for STA130?',
       '4390111: Midway Question 02\r\nWhat are ChatBots helpful for?',
       '4628513: Midway Question 05\r\nWhich of the following best characterizes your experience with the STA130 Custom NotebookLM ChatBot (STA130NBLMbot)?',
       '4390112: Midway Question 03\r\nWhich of the following do you think should continue to be a part of (or be added to) STA130?',
       '4390113: Midway Question 04\r\nWould you be interested in seeing your preferred ChatBots in other STA courses beyond STA130?',
       '4628572: Midway Question 06\r\nNow that you've seen half of the course, completed the midterm exam, and gotten a sense for what Prof. Scott is trying to do with everything in the course... which of the following would you say apply to you?',
       '4

In [27]:
wk07.rename(columns={
    wk07.columns[0]: 'other_chatbots',
    wk07.columns[1]: 'which_chatbots',
    wk07.columns[2]: 'chatbots_good_for',
    wk07.columns[3]: 'experience_with_notebooklm',
    wk07.columns[4]: 'part_of_sta130_doing_good',
    wk07.columns[5]: 'preferred_chatbots_other_sta_course',
    wk07.columns[6]: 'feeling_about_chatbots_sta130',
    wk07.columns[7]: 'worry_about_r_future_sta_course',
    wk07.columns[8]: 'additional_feedback'
}, inplace=True)


# wk07.head()


In [28]:
wk09 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_09_survey.csv')
# wk09.columns

In [29]:
wk09.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','n correct', 'n incorrect','score'], inplace=True)
# wk09.columns

In [30]:
wk09.rename(columns={
    wk09.columns[0]: 'which_chatbots',
    wk09.columns[1]: 'feedback_target',
    wk09.columns[2]: 'five_star_scale',
    wk09.columns[3]: 'learning_support',
    wk09.columns[4]: 'focus',
    wk09.columns[5]: 'additional_feedback',
    wk09.columns[6]: 'feeling_about_chatbots_sta130'
}, inplace=True)

# wk09.head()

In [31]:
wk10 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_10_survey.csv')

wk10.drop(columns=['name','id', 'sis_id', 'section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','1.0.7','n correct', 'n incorrect','score'], inplace=True)

wk10.columns

Index(['4510550: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '4510552: Question 02 (Week 10)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n"My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '4510553: Question 03 (Accuracy)\r\nDid the ChatBot provide helpful guidance and working code?',
       '4510554: Question 04 (Troubleshooting)\r\nIs the ChatBot able to fix programming problemsin two or fewer prompts based on code error messages?',
       '4510555: Question 05 (Focus)\r\nWas the scope of the responses of the ChatBot ap

In [32]:
new_column_names = [
    'which_chatbots',  # Column 0
    'feedback_target',  # Column 1
    'accuracy',  # Column 2
    'troubleshooting',  # Column 3
    'focus',  # Column 4
    'feeling_about_chatbots_sta130',  # Column 5
    'additional_feedback',  # Column 6
    'other_chatbots'  # Column 7
]

# Rename the columns using index positions
wk10.columns = new_column_names

In [33]:
wk11 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_11_survey.csv')

wk11.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','1.0.6','1.0.7','n correct', 'n incorrect','score'], inplace=True)

wk11.columns

Index(['4484890: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '4484891: Question 02 (Week 11)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n"My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '4484894: Question 03 (Five-Star Rating)\r\nRate your experience using a ChatBot in STA130 on a Five-Star scale (where 1 is negative and 5 is positive).\r\nDid you have a good experience regarding ChatBot support in writing, using, and troubleshooting Python code; or, learning statistical, data science, and Python programming topics and

In [34]:
new_column_names = [
    'which_chatbots',  # Column 0
    'feedback_target',  # Column 1
    'five_star_scale',  # Column 2
    'learning_support',  # Column 3
    'focus',  # Column 4
    'feeling_about_chatbots_sta130',  # Column 5
    'additional_feedback',  # Column 6
    'other_chatbots'  # Column 7
]

# Rename the columns using index positions
wk11.columns = new_column_names

In [35]:
wk12 = pd.read_csv('weekly_surveys_raw/STA130_2024_week_12_survey.csv')

wk12.drop(columns=['section','section_id','section_sis_id', 
                          'submitted','attempt','1.0','1.0.1','1.0.2', 
                          '1.0.3','1.0.4','1.0.5','n correct', 'n incorrect','score'], inplace=True)

wk12.columns

Index(['4484884: Question 01\r\nWhich ChatBot(s) do you tend to use consistently and generally find helpful or useful for STA130?',
       '4484885: Question 02 (Project Week 12)\r\nWhich of these topics will your feedback address? And are your responses based on using ChatGPT or the STA130 custom NBLMChatBot?\r\n\r\n"My feedback will be about my experience using a ChatBot for the purposes of..."\r\n\r\nWe're just looking for you to highlight particular aspects of your experience that stood out to you. The options here might not be perfect, but we're hoping they provide enough expressivity for you to give us feedback where it makes sense to you.\r\n✌️🤞',
       '4484887: Question 03 (Accuracy)\r\nDid the ChatBot provide helpful guidance and working code?',
       '4484888: Question 04 (Troubleshooting)\r\nIs the ChatBot able to fix programming problemsin two or fewer prompts based on code error messages?',
       '4484886: Question 05 (Focus)\r\nWas the scope of the responses of the Ch

In [36]:
new_column_names = [
    'which_chatbots',  # Column 0
    'feedback_target',  # Column 1
    'accuracy',  # Column 2
    'troubleshooting',  # Column 3
    'focus',  # Column 4
    'additional_feedback',  # Column 5
]

# Rename the columns using index positions
wk12.columns = new_column_names

## F24 Weekly Surveys

In [37]:
dataframes = [
    ('orientation', orientation),
    ('wk02', wk02),
    ('wk03', wk03),
    ('wk04', wk04),
    ('wk05', wk05),
    ('wk05_after_midterm', wk05_after_midterm),
    ('wk07', wk07),
    ('wk09', wk09),
    ('wk10', wk10),
    ('wk11', wk11),
    ('wk12', wk12)
]

# Add a column to each dataframe indicating its source
for name, df in dataframes:
    df['source'] = name

# Concatenate all dataframes
F24_weekly_surveys = pd.concat([df for _, df in dataframes], ignore_index=True)

# Save the concatenated dataframe to a CSV file
F24_weekly_surveys.to_csv('weekly_surveys_cleaned/F24_weekly_surveys.csv', index=False)

In [50]:
F24_pre.drop(columns=['Id', 'Start time', 'Completion time'], inplace=True)
# Define the text to filter out (the "No..." answer)
text_to_remove = "No, I do not want my data included in any research publications, but I want to complete the surveys. I understand that I will be compensated 0.5% for completing each of the pre- and post-course ChatBots surveys (for a total compensation of 1% of the overall STA130 course grade). By choosing this option I understand that data collected about me may still be used for the purposes of course feedback, but that it will not be used in any way as part of published study results. I also understand that this choice will in no way affect the way I am graded in STA130."

# Filter the DataFrame to exclude rows with the "No..." answer
F24_pre = F24_pre.loc[F24_pre['Please select one of the following:'] != text_to_remove]


In [51]:
F24_pre.rename(columns={'Please select one of the following:': 'concent',
       'Are you an international student?':'international',
                        'Is English your first language?':'ELL',
       'Please tick all that apply. You are:':'race',
       'What first-year calculus course are you currently taking?': 'calculus_course',
       'What first-year programming course are you currently taking?':'cs_course',
       'In what statistics program of study have you enrolled or would you like to enrol in (select your first choice)?':'sta_program',
       'How would you describe your professional writing ability?':'professional_writing_ability',
       'What is your experience with programming prior to starting university in September?':'programming_exprience',
       'What is your experience with PYTHON prior to starting university in September?\n':'python_excprience',
       'How do you evaluate your skill level at PYTHON programming?':'python_skill',
       F24_pre.columns[11]:'sta_skill',
       F24_pre.columns[12]:'sta_in_python_skill',
       F24_pre.columns[13]:'feeling_for_python_in_sta',
       'How much concern do you have that STA130 will not provide enough structure and support for PYTHON?':'concerns',
       'How comfortable are you working independently on your own to understand and use PYTHON code?':'use_python_independently',
       'How interested are you in working with AI assistance as part of learning?':'feeling_AI',
       'How comfortable are you with self-learning statistical concepts and topics?':'self-learning'
}, inplace=True)

In [54]:
F24_post.drop(columns=['Id', 'Start time', 'Completion time','Email', 'Name'], inplace=True)


In [ ]:
# now F24_pre and F24_post are ready 

## what we have now

In [39]:
df_2023_short_col_name
F24_weekly_surveys
F24_post = pd.read_csv('F24_pre_post_cleaned/F24_post_survey_data.csv')
F24_pre = pd.read_excel('F24_pre_post_raw/STA130_Pre_survey_Fall_2024.xlsx')

In [40]:

# 1. python is used in other course;
# 2. stats close to cs but not cs;
# 3. chat also help professional writing too!

# see down below 

In [41]:
df_2023_short_col_name['programming_course'].value_counts()
# python used in 108 and 148, it's good if we can indtroduce stats in same language. 
# + python is useful!!! 

programming_course
CSC108 (or equivalent programming experience)    375
None of the above                                 98
CSC110                                            34
CSC148                                            26
Name: count, dtype: int64

In [42]:
df_2023_short_col_name['statistics_program'].value_counts()

# most student is stats major or specialist, not DS or CS. 
# we want student at least can start use code (with AI help)
# even if we want student can really code individuly, 
# coding with AI help still give student a good/easy start,
# well, good start is good :)

statistics_program
Statistics Major                               254
Data Science Specialist                         82
None of the above                               79
Statistics Minor                                43
Actuarial Science Major                         42
Actuarial Science Specialist                    15
Statistics Specialist: Methods and Practice     14
Statistics Specialist: Theory and Methods        4
Name: count, dtype: int64

In [ ]:
# here are data for result parts 

In [43]:
# Count the number of occurrences of "Good"
count_good = df_2023_short_col_name['pre_professional_writing_ability'].value_counts().get('Good', 0)

# Total number of rows in the column (excluding NaN values)
total_count = df_2023_short_col_name['pre_professional_writing_ability'].notna().sum()

# Calculate the percentage
percentage_good_pre = (count_good / total_count) * 100

# Display the result
print(f"Percentage of 'Good': {percentage_good_pre:.2f}%")


Percentage of 'Good': 40.15%


In [44]:
# Count the number of occurrences of "Good"
count_good = df_2023_short_col_name['post_professional_writing_ability'].value_counts().get('Good', 0)

# Total number of rows in the column (excluding NaN values)
total_count = df_2023_short_col_name['post_professional_writing_ability'].notna().sum()

# Calculate the percentage
percentage_good_post = (count_good / total_count) * 100

# Display the result
print(f"Percentage of 'Good': {percentage_good_post:.2f}%")

Percentage of 'Good': 40.42%


In [59]:
# here we will check the 2024 pre and post professional_writing_ability

 # Count the number of occurrences of "Good"
count_good = F24_pre['professional_writing_ability'].value_counts().get('Good', 0)

# Total number of rows in the column (excluding NaN values)
total_count = F24_pre['professional_writing_ability'].notna().sum()

# Calculate the percentage
percentage_good_post = (count_good / total_count) * 100

# Display the result
print(f"Percentage of 'Good': {percentage_good_post:.2f}%")

Percentage of 'Good': 37.59%


In [64]:
 # Count the number of occurrences of "Good"
count_good = F24_post['how_good_at_after_STA130_writing_for_statisticians'].value_counts().get('Extremely Well', 0)
count_good +=F24_post['how_good_at_after_STA130_writing_for_statisticians'].value_counts().get('Reasonably Well', 0)
# Total number of rows in the column (excluding NaN values)
total_count = F24_post['how_good_at_after_STA130_writing_for_statisticians'].notna().sum()

# Calculate the percentage
percentage_good_post = (count_good / total_count) * 100

# Display the result
print(f"Percentage of 'Good': {percentage_good_post:.2f}%")

Percentage of 'Good': 53.70%







# We can argue chatbot help with professional writing
# in 2023 there is no change in professional_writing_ability 40.15% to 40.42%
# in 2024 there is significant change in professional_writing_ability 37.59% to 53.70%













In [47]:
# we will also see 2024 python_skill_level pre and post 

In [49]:
df_2023_short_col_name['pre_python_skill_level'].value_counts()

pre_python_skill_level
Beginner        409
Intermediate    105
Advanced         14
Other             3
Expert            2
Name: count, dtype: int64

In [48]:
df_2023_short_col_name['post_python_skill_level'].value_counts() 
# still feel struggle with python after fall semaster (after STA130, CSC108 or other cs course. no good. )

post_python_skill_level
Beginner        139
Intermediate     87
Expert           12
Advanced          2
Name: count, dtype: int64

In [71]:
F24_pre['python_skill'].value_counts()

python_skill
Beginner        295
Intermediate     85
Advanced         19
Name: count, dtype: int64

In [73]:
F24_post['how_good_at_after_STA130_using_python'].value_counts()

how_good_at_after_STA130_using_python
Reasonably Well    83
Neutral            67
Extremely Well     40
Not very well      17
Not at all          5
I don't know        4
Name: count, dtype: int64

In [57]:
F24_pre.columns


Index(['concent', 'international', 'ELL', 'race', 'calculus_course',
       'cs_course', 'sta_program', 'professional_writing_ability',
       'programming_exprience', 'python_excprience', 'python_skill',
       'sta_skill', 'sta_in_python_skill', 'feeling_for_python_in_sta',
       'concerns', 'use_python_independently', 'feeling_AI', 'self-learning'],
      dtype='object')

In [56]:
F24_post.columns


Index(['Beyond the above consent decision, we now additionally request your further consent decision allowing us to collect and use the text of "ChatGPT transcript history logs" along with associated HW answe',
       'avg_weekly_hours_on_STA130', '5star_wiki_textbook',
       '5star_custom_notebookLM_chatbot', '5star_lectures', '5star_tutorials',
       '5star_structure', '5star_homework', '5star_project',
       '5star_overall_chatbots',
       ...
       'how_good_at_after_STA130_using_python',
       'how_good_at_after_STA130_writing_for_statisticians',
       'how_good_at_after_STA130_speaking_for_statisticians',
       'how_good_at_after_STA130_translating_statistical_problems',
       'how_good_at_after_STA130_ethics_importance',
       'how_good_at_after_STA130_solving_interdisciplinary_problems',
       'how_good_at_after_STA130_statistical_problem_solving',
       'how_good_at_after_STA130_evaluating_solutions',
       'most_interesting_part_of_course', 'what_to_add_or_chang